library/src/methods/brand/brand.test.ts

```diff
@@ -31,9 +32,51 @@ describe('brand', () => {
 
   test('should allow multiple branding', () => {
     const branded1 = brand(string(), '1');
+    type Branded1 = Output<typeof branded1>;
     const branded2 = brand(branded1, '2');
     type Branded2 = Output<typeof branded2>;
-    expectTypeOf(branded1).not.toEqualTypeOf(branded2);
+
+    expectTypeOf<Branded1>().not.toBeNever();
+    expectTypeOf<Branded2>().not.toBeNever();
+
+    expectTypeOf<Branded2>().toMatchTypeOf<Branded1>();
+    expectTypeOf<Branded1>().not.toMatchTypeOf<Branded2>();
+
     expectTypeOf<Branded2>().toEqualTypeOf<string & Brand<'1'> & Brand<'2'>>();
+    expectTypeOf<Branded2>().toMatchTypeOf<string & Brand<'1'>>();
+  });
+
+  test('should support brand hierarchy', () => {
+    const id = brand(string(), 'Id');
+    type BrandedId = Output<typeof id>;
+    const userId = brand(id, 'UserId');
+    type BrandedUserId = Output<typeof userId>;
+    const postId = brand(id, 'PostId');
+    type BrandedPostId = Output<typeof postId>;
+
+    expectTypeOf<BrandedId>().not.toBeNever();
+    expectTypeOf<BrandedUserId>().not.toBeNever();
+    expectTypeOf<BrandedPostId>().not.toBeNever();
+
+    expectTypeOf<BrandedUserId>().not.toMatchTypeOf<BrandedPostId>();
+    expectTypeOf<BrandedPostId>().not.toMatchTypeOf<BrandedUserId>();
+  });
+
+  test('should support unbranding', () => {
```

Comment:
Why is unbranding and this test necessary? Is there a practical use case for this?

Reply:
Can you explain the practical use case? Does Zod have this feature?